metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1572,7 +1583,7 @@ class MetalsLanguageServer(
       targets.asScala.foreach { target =>
         buildTargets.linkSourceFile(target, source)
       }
-      indexSourceFile(source, Some(sourceItem))
+      indexSourceFile(source, Some(sourceItem), targets.asScala.headOption)
```

Comment:
Maybe we could have this:
```scala
    val toIndexSource = targetOpt
      .flatMap(ammonite.generatedScalaPath(_, source))
      .getOrElse(source)
```
in a separate function and just use here to get the real source item?

Reply:
Done.